r74891 | ascherer | 2025-04-09 12:41:10 +0200 (Wed, 09 Apr 2025) | 3 lines
[CWEB] Revert bool change for C23.
Plain CWEB 4 uses <stdbool.h> all along.

--- texlive-bin-2025.20250325.74682.orig/texk/web2c/ctangleboot.cin
+++ texlive-bin-2025.20250325.74682/texk/web2c/ctangleboot.cin
@@ -2,10 +2,10 @@
 #line 66 "cwebdir/ctangle.w"
 
 /*5:*/
-#line 57 "cwebdir/comm-w2c.h"
+#line 54 "cwebdir/comm-w2c.h"
 
 #include <ctype.h>  
-#include <kpathsea/simpletypes.h>  
+#include <stdbool.h>  
 #include <stddef.h>  
 #include <stdint.h>  
 #include <stdio.h>  
@@ -15,15 +15,15 @@
 #ifndef HAVE_GETTEXT
 #define HAVE_GETTEXT 0
 #endif
-#line 69 "cwebdir/comm-w2c.h"
+#line 66 "cwebdir/comm-w2c.h"
 
 #if HAVE_GETTEXT
 #include <libintl.h> 
 #else
-#line 73 "cwebdir/comm-w2c.h"
+#line 70 "cwebdir/comm-w2c.h"
 #define gettext(a) a
 #endif
-#line 75 "cwebdir/comm-w2c.h"
+#line 72 "cwebdir/comm-w2c.h"
 
 /*:5*/
 #line 67 "cwebdir/ctangle.w"
@@ -177,8 +177,9 @@
 #line 68 "cwebdir/ctangle.w"
 
 /*3:*/
-#line 33 "cwebdir/comm-w2c.h"
+#line 32 "cwebdir/comm-w2c.h"
 
+typedef bool boolean;
 typedef uint8_t eight_bits;
 typedef uint16_t sixteen_bits;
 typedef enum{
@@ -188,7 +189,7 @@
 extern int phase;
 
 /*:3*//*6:*/
-#line 97 "cwebdir/comm-w2c.h"
+#line 94 "cwebdir/comm-w2c.h"
 
 extern char section_text[];
 extern char*section_text_end;
@@ -196,7 +197,7 @@
 extern char*id_loc;
 
 /*:6*//*7:*/
-#line 115 "cwebdir/comm-w2c.h"
+#line 112 "cwebdir/comm-w2c.h"
 
 extern char buffer[];
 extern char*buffer_end;
@@ -204,7 +205,7 @@
 extern char*limit;
 
 /*:7*//*8:*/
-#line 132 "cwebdir/comm-w2c.h"
+#line 129 "cwebdir/comm-w2c.h"
 
 extern int include_depth;
 extern FILE*file[];
@@ -221,7 +222,7 @@
 extern boolean web_file_open;
 
 /*:8*//*10:*/
-#line 153 "cwebdir/comm-w2c.h"
+#line 150 "cwebdir/comm-w2c.h"
 
 extern sixteen_bits section_count;
 extern boolean changed_section[];
@@ -229,7 +230,7 @@
 extern boolean print_where;
 
 /*:10*//*11:*/
-#line 168 "cwebdir/comm-w2c.h"
+#line 165 "cwebdir/comm-w2c.h"
 
 typedef struct name_info{
 char*byte_start;
@@ -254,12 +255,12 @@
 extern hash_pointer hash_ptr;
 
 /*:11*//*13:*/
-#line 213 "cwebdir/comm-w2c.h"
+#line 210 "cwebdir/comm-w2c.h"
 
 extern int history;
 
 /*:13*//*15:*/
-#line 234 "cwebdir/comm-w2c.h"
+#line 231 "cwebdir/comm-w2c.h"
 
 extern int argc;
 extern char**argv;
@@ -272,7 +273,7 @@
 extern const char*use_language;
 
 /*:15*//*16:*/
-#line 250 "cwebdir/comm-w2c.h"
+#line 247 "cwebdir/comm-w2c.h"
 
 extern FILE*C_file;
 extern FILE*tex_file;
@@ -395,14 +396,14 @@
 extern void common_init(void);
 
 /*:4*//*9:*/
-#line 147 "cwebdir/comm-w2c.h"
+#line 144 "cwebdir/comm-w2c.h"
 
 extern boolean get_line(void);
 extern void check_complete(void);
 extern void reset_input(void);
 
 /*:9*//*12:*/
-#line 191 "cwebdir/comm-w2c.h"
+#line 188 "cwebdir/comm-w2c.h"
 
 extern name_pointer id_lookup(const char*,const char*,eight_bits);
 
@@ -416,7 +417,7 @@
 extern void init_node(name_pointer);
 
 /*:12*//*14:*/
-#line 216 "cwebdir/comm-w2c.h"
+#line 213 "cwebdir/comm-w2c.h"
 
 extern int wrap_up(void);
 extern void err_print(const char*);
--- texlive-bin-2025.20250325.74682.orig/texk/web2c/cwebboot.cin
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebboot.cin
@@ -2,10 +2,10 @@
 #line 62 "cwebdir/common.w"
 
 /*4:*/
-#line 57 "cwebdir/comm-w2c.h"
+#line 54 "cwebdir/comm-w2c.h"
 
 #include <ctype.h>  
-#include <kpathsea/simpletypes.h>  
+#include <stdbool.h>  
 #include <stddef.h>  
 #include <stdint.h>  
 #include <stdio.h>  
@@ -15,15 +15,15 @@
 #ifndef HAVE_GETTEXT
 #define HAVE_GETTEXT 0
 #endif
-#line 69 "cwebdir/comm-w2c.h"
+#line 66 "cwebdir/comm-w2c.h"
 
 #if HAVE_GETTEXT
 #include <libintl.h> 
 #else
-#line 73 "cwebdir/comm-w2c.h"
+#line 70 "cwebdir/comm-w2c.h"
 #define gettext(a) a
 #endif
-#line 75 "cwebdir/comm-w2c.h"
+#line 72 "cwebdir/comm-w2c.h"
 
 /*:4*//*91:*/
 #line 739 "cwebdir/comm-w2c.ch"
@@ -41,13 +41,15 @@
 /*:91*//*93:*/
 #line 786 "cwebdir/comm-w2c.ch"
 
+typedef bool boolean;
+#define HAVE_BOOLEAN 1
 #include <kpathsea/kpathsea.h>  
 
 #include <w2c/config.h>  
 #include <lib/lib.h>  
 
 /*:93*//*96:*/
-#line 823 "cwebdir/comm-w2c.ch"
+#line 825 "cwebdir/comm-w2c.ch"
 
 #define CWEB
 #include "help.h" 
@@ -179,8 +181,9 @@
 #line 64 "cwebdir/common.w"
 
 /*2:*/
-#line 33 "cwebdir/comm-w2c.h"
+#line 32 "cwebdir/comm-w2c.h"
 
+typedef bool boolean;
 typedef uint8_t eight_bits;
 typedef uint16_t sixteen_bits;
 typedef enum{
@@ -190,7 +193,7 @@
 extern int phase;
 
 /*:2*//*5:*/
-#line 97 "cwebdir/comm-w2c.h"
+#line 94 "cwebdir/comm-w2c.h"
 
 extern char section_text[];
 extern char*section_text_end;
@@ -198,7 +201,7 @@
 extern char*id_loc;
 
 /*:5*//*6:*/
-#line 115 "cwebdir/comm-w2c.h"
+#line 112 "cwebdir/comm-w2c.h"
 
 extern char buffer[];
 extern char*buffer_end;
@@ -206,7 +209,7 @@
 extern char*limit;
 
 /*:6*//*7:*/
-#line 132 "cwebdir/comm-w2c.h"
+#line 129 "cwebdir/comm-w2c.h"
 
 extern int include_depth;
 extern FILE*file[];
@@ -223,7 +226,7 @@
 extern boolean web_file_open;
 
 /*:7*//*9:*/
-#line 153 "cwebdir/comm-w2c.h"
+#line 150 "cwebdir/comm-w2c.h"
 
 extern sixteen_bits section_count;
 extern boolean changed_section[];
@@ -231,7 +234,7 @@
 extern boolean print_where;
 
 /*:9*//*10:*/
-#line 168 "cwebdir/comm-w2c.h"
+#line 165 "cwebdir/comm-w2c.h"
 
 typedef struct name_info{
 char*byte_start;
@@ -256,12 +259,12 @@
 extern hash_pointer hash_ptr;
 
 /*:10*//*12:*/
-#line 213 "cwebdir/comm-w2c.h"
+#line 210 "cwebdir/comm-w2c.h"
 
 extern int history;
 
 /*:12*//*14:*/
-#line 234 "cwebdir/comm-w2c.h"
+#line 231 "cwebdir/comm-w2c.h"
 
 extern int argc;
 extern char**argv;
@@ -274,7 +277,7 @@
 extern const char*use_language;
 
 /*:14*//*15:*/
-#line 250 "cwebdir/comm-w2c.h"
+#line 247 "cwebdir/comm-w2c.h"
 
 extern FILE*C_file;
 extern FILE*tex_file;
@@ -427,14 +430,14 @@
 extern void common_init(void);
 
 /*:3*//*8:*/
-#line 147 "cwebdir/comm-w2c.h"
+#line 144 "cwebdir/comm-w2c.h"
 
 extern boolean get_line(void);
 extern void check_complete(void);
 extern void reset_input(void);
 
 /*:8*//*11:*/
-#line 191 "cwebdir/comm-w2c.h"
+#line 188 "cwebdir/comm-w2c.h"
 
 extern name_pointer id_lookup(const char*,const char*,eight_bits);
 
@@ -448,7 +451,7 @@
 extern void init_node(name_pointer);
 
 /*:11*//*13:*/
-#line 216 "cwebdir/comm-w2c.h"
+#line 213 "cwebdir/comm-w2c.h"
 
 extern int wrap_up(void);
 extern void err_print(const char*);
@@ -488,7 +491,7 @@
 
 #line 514 "cwebdir/comm-w2c.ch"
 /*:76*//*98:*/
-#line 836 "cwebdir/comm-w2c.ch"
+#line 838 "cwebdir/comm-w2c.ch"
 
 static void cb_usage(const_string str);
 static void cb_usagehelp(const_string*message);
@@ -516,7 +519,7 @@
 #line 78 "cwebdir/comm-w2c.ch"
 
 /*94:*/
-#line 798 "cwebdir/comm-w2c.ch"
+#line 800 "cwebdir/comm-w2c.ch"
 
 kpse_set_program_name(argv[0],"cweb");
 
@@ -1442,7 +1445,7 @@
 if(strcmp("-help",*argv)==0||strcmp("--help",*argv)==0)
 
 /*97:*/
-#line 827 "cwebdir/comm-w2c.ch"
+#line 829 "cwebdir/comm-w2c.ch"
 
 cb_usagehelp(program==ctangle?CTANGLEHELP:
 program==cweave?CWEAVEHELP:CTWILLHELP);
@@ -1454,7 +1457,7 @@
 if(strcmp("-version",*argv)==0||strcmp("--version",*argv)==0)
 
 /*100:*/
-#line 873 "cwebdir/comm-w2c.ch"
+#line 875 "cwebdir/comm-w2c.ch"
 
 printversionandexit(cb_banner,
 program==ctwill?"Donald E. Knuth":"Silvio Levy and Donald E. Knuth",
@@ -1628,7 +1631,7 @@
 }
 
 /*:75*//*99:*/
-#line 840 "cwebdir/comm-w2c.ch"
+#line 842 "cwebdir/comm-w2c.ch"
 
 static void cb_usage(const_string str)
 {
@@ -1660,7 +1663,7 @@
 }
 
 /*:99*//*101:*/
-#line 881 "cwebdir/comm-w2c.ch"
+#line 883 "cwebdir/comm-w2c.ch"
 
 void cb_show_banner(void)
 {
--- texlive-bin-2025.20250325.74682.orig/texk/web2c/cwebdir/comm-w2c.ch
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w2c.ch
@@ -784,6 +784,8 @@
 @d kpse_find_cweb(name) kpse_find_file(name,kpse_cweb_format,true)
 
 @<Include files@>=
+typedef bool boolean;
+#define HAVE_BOOLEAN 1
 #include <kpathsea/kpathsea.h> /* include every \Kpathsea/ header;
   |@!kpathsea_debug|, |@!const_string|, |@!string| */
 #include <w2c/config.h> /* \&{integer} */
--- texlive-bin-2025.20250325.74682.orig/texk/web2c/cwebdir/comm-w2c.h
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/comm-w2c.h
@@ -29,8 +29,8 @@
 
 @i iso_types.w
 
-@s boolean bool
 @<Common code...@>=@^system dependencies@>
+typedef bool boolean;
 typedef uint8_t eight_bits;
 typedef uint16_t sixteen_bits;
 typedef enum {
@@ -48,15 +48,12 @@
 `|@!gettext|' function from the ``GNU~gettext utilities.'' For systems that do
 not have this library installed, we wrap things for neutral behavior without
 internationalization.
-For backward compatibility with pre-{\mc ANSI} compilers, we replace the
-``standard'' header file `\.{stdbool.h}' with the@^system dependencies@>
-{\mc KPATHSEA\spacefactor1000} interface `\.{simpletypes.h}'.
 
 @d _(s) gettext(s)
 
 @<Include files@>=
 #include <ctype.h> /* definition of |@!isalpha|, |@!isdigit| and so on */
-#include <kpathsea/simpletypes.h> /* |@!boolean|, |@!true| and |@!false| */
+#include <stdbool.h> /* definition of |@!bool|, |@!true| and |@!false| */
 #include <stddef.h> /* definition of |@!ptrdiff_t| */
 #include <stdint.h> /* definition of |@!uint8_t| and |@!uint16_t| */
 #include <stdio.h> /* definition of |@!printf| and friends */
--- texlive-bin-2025.20250325.74682.orig/texk/web2c/cwebdir/ctwill-mini.ch
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-mini.ch
@@ -88,12 +88,6 @@
 Section 5.
 
 @x
-For backward compatibility with pre-{\mc ANSI} compilers, we replace the
-@y
-And we replace the
-@z
-
-@x
 @d _(s) gettext(s)
 @y
 @d _(s) gettext(s)
@@ -1657,9 +1651,12 @@
 Section 215.
 
 @x
-@<Append a \9{s}string or...@>={@+ int count=-1; /* characters remaining before string break */
+  if (count==0) { /* insert a discretionary break in a long string */
+     app_str(@q(@>@q{@>"}\\)\\.{"@q}@>); count=20;
 @y
-@<Append a \9{s}string or...@>={@+ int count=-1; /* characters remaining before string break */
+  if (count==0) {
+     count=20; @/ /* insert a discretionary break in a long string */
+     app_str(@q(@>@q{@>"}\\)\\.{"@q}@>);
 @z
 
 Section 220.
@@ -1903,6 +1900,14 @@
 @ @<Translate the \9{c}current section@>= @+ {
 @z
 
+Section 249.
+
+@x
+    case section_name: loc-=2; next_control=get_next(); /* reprocess */
+@y
+    case section_name: @/ loc-=2; @/ next_control=get_next(); /* reprocess */
+@z
+
 Section 251.
 
 @x
@@ -2327,6 +2332,7 @@
 @y
 @d kpse_find_cweb(name) kpse_find_file(name,kpse_cweb_format,true)
 @-name@>
+@-HAVE_BOOLEAN@>
 @z
 
 Section 329.
--- texlive-bin-2025.20250325.74682.orig/texk/web2c/cwebdir/ctwill-w2c.ch
+++ texlive-bin-2025.20250325.74682/texk/web2c/cwebdir/ctwill-w2c.ch
@@ -2009,6 +2009,8 @@
 @d kpse_find_cweb(name) kpse_find_file(name,kpse_cweb_format,true)
 
 @<Include files@>=
+typedef bool boolean;
+#define HAVE_BOOLEAN 1
 #include <kpathsea/tex-file.h> /* |@!kpse_find_file| */
 
 @ @<Set up |cur_file_name|...@>=
